profiles/common/modules/custom/tmgmt_workbench/tmgmt_workbench.module

```diff
@@ -250,10 +251,22 @@ function tmgmt_workbench_form_tmgmt_entity_ui_translate_form_alter(&$form, $form
  *   TRUE if sending a request is possible otherwise FALSE.
  */
 function _tmgmt_workbench_check_translatability($entity_id) {
-  $main_job = _tmgmt_poetry_get_main_job_by_entity_id($entity_id);
-  $translator = $main_job->getTranslator();
+  // Getting the main TMGMT job based on the given entity id.
+  $query = db_select('tmgmt_job_item', 'tji');
+  $query->join('tmgmt_job', 'tj', 'tji.tjid = tj.tjid');
+  $query->fields('tj', ['tjid', 'reference']);
+  $query->condition('tji.item_id', $entity_id, '=');
+  $query->condition('reference', "^(MAIN)_", "REGEXP");
```

Comment:
This is still depending quite a lot on tmgmt_poetry, because you are looking for a reference starting with MAIN_ which is quite poetry specific.
Also, tmgmt_workbench_add_target_languages_form_submit() is still calling _tmgmt_poetry_get_main_job_by_entity_id() thus the intertangled dependency is not really solved yet.